content/backend/graphql-scala/0-introduction.md

```diff
@@ -0,0 +1,174 @@
+2. Implement functions called **resolvers** to handle these types and their fields.
+3. As new requirements arrive, go back to step 1 to update the schema, and continue through the other steps.
+
+The schema is a contract agreed on between the frontend and backend, so keeping it at the center allows both sides of the development to evolve without going off the spec. This also makes it easier to parallelize the work, since the frontend can move on with full knowledge of the API from the start, using a simple mocking service (or even a full backend such as Graphcool) which can later be easily replaced with the final server.
+
+
+### Goal of the turorial
+
+The goal of this tutorial is to make server able to run the following schema, :
+
+
+```graphql(nocopy)(https://github.com/howtographql/howtographql/blob/master/meta/structure.graphql)
+type Query {
+  allLinks(filter: LinkFilter, orderBy: LinkOrderBy, skip: Int, first: Int): [Link!]!
+  _allLinksMeta: _QueryMeta!
+}
+
+type Mutation {
+  signinUser(email: AUTH_PROVIDER_EMAIL): SigninPayload!
+  createUser(name: String!, authProvider: AuthProviderSignupData!): User
+  createLink(description: String!, url: String!, postedById: ID): Link
+  createVote(linkId: ID, userId: ID): Vote
+}
+
+type Subscription {
+  Link(filter: LinkSubscriptionFilter): LinkSubscriptionPayload
+  Vote(filter: VoteSubscriptionFilter): VoteSubscriptionPayload
+}
+
+interface Node {
+  id: ID!
+}
+
+type User implements Node {
+  id: ID! @isUnique
+  createdAt: DateTime!
+  name: String!
+  links: [Link!]! @relation(name: "UsersLinks")
+  votes: [Vote!]! @relation(name: "UsersVotes")
+  email: String @isUnique
+  password: String
+}
+
+type Link implements Node {
+  id: ID! @isUnique
+  createdAt: DateTime!
+  url: String!
+  description: String!
+  postedBy: User! @relation(name: "UsersLinks")
+  votes: [Vote!]! @relation(name: "VotesOnLink")
+}
+
+type Vote implements Node {
+  id: ID! @isUnique
+  createdAt: DateTime!
+  user: User! @relation(name: "UsersVotes")
+  link: Link! @relation(name: "VotesOnLink")
+}
+
+input AuthProviderSignupData {
+  email: AUTH_PROVIDER_EMAIL
+}
+
+input AUTH_PROVIDER_EMAIL {
+  email: String!
+  password: String!
+}
+
+input LinkSubscriptionFilter {
+  mutation_in: [_ModelMutationType!]
+}
+
+input VoteSubscriptionFilter {
+  mutation_in: [_ModelMutationType!]
+}
+
+input LinkFilter {
+  OR: [LinkFilter!]
+  description_contains: String
+  url_contains: String
+}
+
+type SigninPayload {
+  token: String
+  user: User
+}
+
+type LinkSubscriptionPayload {
+  mutation: _ModelMutationType!
+  node: Link
+  updatedFields: [String!]
+}
+
+type VoteSubscriptionPayload {
+  mutation: _ModelMutationType!
+  node: Vote
+  updatedFields: [String!]
+}
+
+enum LinkOrderBy {
+  createdAt_ASC
+  createdAt_DESC
+}
+
+enum _ModelMutationType {
+  CREATED
+  UPDATED
+  DELETED
+}
+
+type _QueryMeta {
+  count: Int!
+}
+
+scalar DateTime
+```
```

Comment:
If the reader got to this point it would be good to reinforce his will to read on.
`Looks complicated? Fear not, GraphQL makes it simple. To find out more, check part 1/2/3 of this tutorial`